integrations/oci/oci-base/src/main/java/io/helidon/integrations/oci/Oci.java

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+
+package io.helidon.integrations.oci;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+
+import io.helidon.config.Config;
+
+import com.oracle.bmc.ClientConfiguration;
+import com.oracle.bmc.Region;
+import com.oracle.bmc.auth.AuthenticationDetailsProvider;
+import com.oracle.bmc.auth.BasicAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.ConfigFileAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.InstancePrincipalsAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.ResourcePrincipalAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.SimpleAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.URLBasedX509CertificateSupplier;
+import com.oracle.bmc.auth.internal.AuthUtils;
+
+
+/**
+ * OCI support for Helidon.
+ */
+public class Oci {
+
+
+    private AuthenticationDetailsProvider provider;
+    private ClientConfiguration clientConfig;
+
+    private Oci() {
+        //private constructor
+    }
+
+    private Oci(Builder builder) {
+        provider = builder.configureProvider();
+        clientConfig = builder.getClientConfig();
+    }
+
+    /**
+     * The OCI create method.
+     *
+     * @param config from SE.
+     * @return using builder pattern.
+     */
+    public static Oci create(Config config) {
+        return builder().config(config).build();
+    }
+
+    /**
+     * Get the registered provider.
+     *
+     * @return AuthenticationDetailsProvider.
+     */
+    public AuthenticationDetailsProvider getProvider() {
```

Comment:
Naming convention: getters and setters do not use verbs, e.g. this should be called `provider()`

Reply:
fixed